WordPress/Classes/ViewRelated/Reader/ReaderSidebarViewController.swift

```diff
@@ -33,15 +33,18 @@ final class ReaderSidebarViewController: UIHostingController<AnyView> {
     func showInitialSelection() {
         cancellables = []
 
-        viewModel.$selection.sink { [weak self] in
-            self?.configure(for: $0)
-        }.store(in: &cancellables)
+        // -warning: List occasionally sets the selection to `nil` when switching items.
+        viewModel.$selection.compactMap { $0 }
+            .removeDuplicates { [weak self] in
+                guard $0 == $1 else { return false }
```

Comment:
I think this is the easiest way to implement it without introducing any additional state.

Reply:
What do you think implementing it by handing double tap gesture?

At the moment, double tap would lead to `selection` getting the same value. But `selection` getting the same value doesn't always mean a double tap.

It's very possible that `selection` is changed to the same value, programmatically rather than by user interaction. For example, when tapping a reader post deep link while the app is already in the Reader, the code may assign `.reader` to the `selection` again.

Reply:
> What do you think implementing it by handing double tap gesture?

It has to be single because that's what Apple apps do.

> programmatically rather than by user interaction

That's a good point. I'm not sure how to address it because the way `List` selection in the sidebar works is also indirect. There is no "user tapped cell X" callback 🤔

Reply:
I think it's OK to ship it this way because if it's a deep link for let's say "Discover", it should be OK to pop to root. Both the deep links and Split View needs to be revised in the future versions.

Reply:
I only checked out Apple Podcasts app. And it has inconsistent UX. If you tap one of the tabs (i.e. Browse), the app takes you to the root. But if you tap other items (like in the Library section), it doesn't. I feel like Apple apps may piggy pack on UITabBarController's implementation.

> I think it's OK to ship it this way because if it's a deep link for let's say "Discover"

 Deep link is just one example on top of my head. From the code level, `viewModel.selection = ...` can be done at any stage, and it's very likely that we would forget this simple assignment may lead to a navigation stack change to the app UI.

I guess it's okay to ship it as the potential issue is limited on the Reader tab only.